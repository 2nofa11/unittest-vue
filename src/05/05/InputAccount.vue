<script setup lang="ts">
const email = defineModel<string>("email");
const password = defineModel<string>("password");
</script>

<template>
  <fieldset data-test="input-account-fieldset">
    <legend data-test="input-account-legend">アカウント情報の入力</legend>
    <div>
      <label>
        メールアドレス
        <input
          type="text"
          placeholder="example@test.com"
          v-model="email"
          data-test="input-email"
        />
      </label>
    </div>
    <div>
      <label>
        パスワード
        <input
          type="password"
          placeholder="8文字以上で入力"
          v-model="password"
          data-test="input-password"
        />
      </label>
    </div>
  </fieldset>
</template>
