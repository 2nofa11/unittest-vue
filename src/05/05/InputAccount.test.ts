import { describe, expect, test } from "@jest/globals";
import { mount } from "@vue/test-utils";
import InputAccount from "./InputAccount.vue";

describe("InputAccount.vue", () => {
  test("fieldset が表示され、legend が含まれる", () => {
    const wrapper = mount(InputAccount, { props: { email: "", password: "" } });
    const fieldset = wrapper.find('[data-test="input-account-fieldset"]');
    expect(fieldset.exists()).toBe(true);
    expect(fieldset.find("legend").text()).toBe("アカウント情報の入力");
  });

  test("メールアドレス入力欄", async () => {
    const wrapper = mount(InputAccount, { props: { email: "", password: "" } });
    const emailInput = wrapper.find<HTMLInputElement>(
      '[data-test="input-email"]'
    );
    const testEmail = "test@example.com";

    await emailInput.setValue(testEmail);

    expect(emailInput.element.value).toBe(testEmail);
  });

  test("パスワード入力欄", async () => {
    const wrapper = mount(InputAccount, { props: { email: "", password: "" } });
    const passwordInput = wrapper.find<HTMLInputElement>(
      '[data-test="input-password"]'
    );
    expect(passwordInput.element.placeholder).toBe("8文字以上で入力");
    const testPassword = "password123";

    await passwordInput.setValue(testPassword);

    expect(passwordInput.element.value).toBe(testPassword);
  });

  test("Snapshot: アカウント情報の入力フォームが表示される", () => {
    const wrapper = mount(InputAccount, { props: { email: "", password: "" } });
    expect(wrapper.html()).toMatchSnapshot();
  });
});
