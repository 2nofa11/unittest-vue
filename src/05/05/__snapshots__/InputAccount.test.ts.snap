// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputAccount.vue Snapshot: アカウント情報の入力フォームが表示される 1`] = `
"<fieldset data-test="input-account-fieldset">
  <legend data-test="input-account-legend">アカウント情報の入力</legend>
  <div><label> メールアドレス <input type="text" placeholder="example@test.com" data-test="input-email"></label></div>
  <div><label> パスワード <input type="password" placeholder="8文字以上で入力" data-test="input-password"></label></div>
</fieldset>"
`;
